### PR TITLE
Cyphers/static

### PR DESCRIPTION
**Pull request by diyessi**
Make static linking work on macs
Remove circular dependencies between backends and ngraph
Simplify backend registration
Many files, not many actual changes.

This PR should be merged after
~~#3514 visibility~~ merged
#3516 cpu changes
which are currently also included in this PR.

### CODE REVIEW

File: changes.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # API Changes
 
+## Backend library interface
+* Each backend `BACKEND` needs to define the macro `${BACKEND}_API` appopriately to import symbols
```
Comment (indie) [inline changes.md]: appropriately

**Comment by earhart** [comment]
Just wondering -- what's the goal here?  (Is there a customer requirement for a static libngraph?)

File: src/ngraph/runtime/generic_cpu/gcpu_backend.cpp [hunk 2]
```diff
@@ -14,30 +14,23 @@
...
+#include "ngraph/runtime/generic_cpu/gcpu_backend.hpp"
 #include "ngraph/runtime/generic_cpu/gcpu_executable.hpp"
 #include "ngraph/runtime/host_tensor.hpp"
 #include "ngraph/util.hpp"
 
 using namespace std;
 using namespace ngraph;
 
-extern "C" runtime::BackendConstructor* get_backend_constructor_pointer()
+extern "C" GCPU_BACKEND_API void ngraph_register_nop_backend()
```
Comment (rkimballn1) [inline src/ngraph/runtime/generic_cpu/gcpu_backend.cpp]: Shoud be `ngraph_register_gcpu_backend()`

File: src/ngraph/runtime/plaidml/unit_test.manifest [hunk 3]
```diff
@@ -39,79 +39,88 @@ topk_2d_min_one                         # No plans to implement TopK
 topk_int64                              # No plans to implement TopK
```
Comment (rkimballn1) [inline src/ngraph/runtime/plaidml/unit_test.manifest]: Why was this file changed so much? Can you revert it?
